### PR TITLE
[examples/seq2seq]: add --label_smoothing option

### LINKED ISSUE #5897: examples/seq2seq: add label_smoothing cross entropy option
Started!

### PR DESCRIPTION
**Pull request by sshleifer**
cc @patil-suraj 
This seems to improve BLEU score by ~2pts!
- also adds --early_stopping_patience command line arg
- fixes MBartDataset src,tgt flipping bug
- adds `--early_stopping_patience` command line arg for PL.
- wandb now looks for shell variable `$WANDB_PROJECT_NAME`

### CODE REVIEW

File: examples/seq2seq/utils.py [hunk 1]
```diff
@@ -19,6 +19,27 @@
...
+        smooth_loss.masked_fill_(pad_mask, 0.0)
+    else:
+        nll_loss = nll_loss.squeeze(-1)
+        smooth_loss = smooth_loss.squeeze(-1)
+    if reduce:
+        nll_loss = nll_loss.sum()  # mean()? Scared to break other math.
+        smooth_loss = smooth_loss.sum()
+    eps_i = epsilon / lprobs.size(-1)
+    loss = (1.0 - epsilon) * nll_loss + eps_i * smooth_loss
+    return loss / lprobs.shape[0], nll_loss / lprobs.shape[0]
```
Comment (sgugger) [inline examples/seq2seq/utils.py:40]: The division should be done by the actual batch size, which should not take the `ignore_index` into account. Using
```
bs = lprobs.shape[0] if ignore_index is None else lprobs.shape[0] - pad_mask.long().sum()
return loss / bs, nll_loss / bs
```
gives the same result as cross entropy for `nll_loss` (and will allow you to uncomment the assert).

**Comment by sshleifer** [comment]
Gunna merge this.
The packed dataset is definitely a win, label smoothing less clear.
TODO: figure out loss function mystery.
